precovery/healpix_geom.py

```diff
@@ -1,9 +1,21 @@
+from typing import overload
+
 import healpy as hp
 import numpy as np
 import numpy.typing as npt
 
 
+@overload
```

Comment:
Do we use the scalar definition of this function fairly often? I'm wondering if we can drop it in favor of just the array-typed one. If it is used, and I suspect that it is when HealpixFrames are initialized then no worries.